src/Microsoft.AspNetCore.Mvc.Core/Internal/MiddlewarePipelineResourceFilter.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    public class MiddlewarePipelineResourceFilter : IAsyncResourceFilter, IOrderedFilter
+    {
+        private readonly RequestDelegate _requestDelegate;
+
+        public MiddlewarePipelineResourceFilter(RequestDelegate requestDelegate)
+        {
+            _requestDelegate = requestDelegate;
+        }
+
+        public int Order
+        {
+            get
+            {
+                return int.MinValue + 100;
+            }
+        }
+
+        public Task OnResourceExecutionAsync(ResourceExecutingContext context, ResourceExecutionDelegate next)
+        {
+            var httpContext = context.HttpContext;
+
+            httpContext.Items.Add(
+                typeof(ResourceFilterContext),
```

Comment:
`typeof(ResourceFilterContext)`? is kinda gross (and public). Make it an interal object (unless we want people to muck with it.